includes/acl/core/memory.h

```diff
@@ -82,6 +82,15 @@ namespace acl
 		return new(ptr) AllocatedType();
 	}
 
+	template<typename AllocatedType, typename... Args>
+	AllocatedType* allocate_type(Allocator& allocator, Args... args)
```

Comment:
good addition, if we pass arguments, we should statis_assert that the type isn't trivially_default_constructible, if it is, the arguments are ignored and something isn't right